server/src/rbx_snapshot.rs

```diff
@@ -202,24 +212,33 @@ pub fn snapshot_project_node<'source>(
 ) -> SnapshotResult<'source> {
     let maybe_snapshot = match &node.path {
         Some(path) => snapshot_imfs_path(context, imfs, &path, Some(instance_name.clone()))?,
-        None => match &node.class_name {
-            Some(_class_name) => Some(RbxSnapshotInstance {
+        None => {
+            let class_name = match &node.class_name {
+                Some(_class_name) => Cow::Borrowed("Folder"),
+                None => {
+                    match infer_class_name(instance_name.clone()) {
+                        Some(inferred_class_name) => inferred_class_name,
+                        None => {
+                            return Err(SnapshotError::ProjectNodeUnusable);
+                        }
+                    }
+                },
```

Comment:
I think we can drop the braces here so that the `None` match cascades right into another `match`.

We _also_ should be able to write the inner match as

```rust
infer_class_name(instance_name.clone()).ok_or(SnapshotError::ProjectNodeUnusable)?
```

so something like

```rust
            let class_name = match &node.class_name {
                Some(_) => Cow::Borrowed("Folder"),
                None => infer_class_name(instance_name.clone())
                    .ok_or(SnapshotError::ProjectNodeUnusable)?,
            }
```